test/functional/api/window_spec.lua

```diff
@@ -889,6 +889,253 @@ describe('API/win', function()
          }))
       eq(wins_before, meths.list_wins())
     end)
+
+    it('creates a split window', function()
+      local win = meths.open_win(0, true, {
+        win = 0,
+        vertical = false
+      })
+      eq(meths.win_get_config(win).relative, '')
+    end)
+
+    it('creates split windows in the correct direction', function()
+      local win = meths.open_win(0, true, {
+        vertical = true,
+        win = 0,
+      })
+      eq(meths.win_get_config(win).relative, '')
+
+      local layout = meths.call_function('winlayout', {})
+
+      -- walk the tree, and ensure the window is a vertical split
+      local queue = { layout }
+      local parent
+      local current
+      while true do
```

Comment:
nit: make this a for-loop that fails after e.g. 1000 iterations, to avoid a hanging test.

Reply:
Good idea, will do